cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -560,3 +571,186 @@ fn register_weth_token_in_asset_hub_fail_for_insufficient_fee() {
+
+	BridgeHubRococo::execute_with(|| {
+		type Runtime = <BridgeHubRococo as Chain>::Runtime;
+
+		let agent_id = snowbridge_pallet_system::agent_id_of::<Runtime>(&Location::new(
+			1,
+			[Parachain(AssetHubRococo::para_id().into())],
+		))
+		.unwrap();
+		snowbridge_pallet_system::Agents::<Runtime>::insert(agent_id, ());
+		let channel_id: ChannelId = AssetHubRococo::para_id().into();
+		snowbridge_pallet_system::Channels::<Runtime>::insert(
+			channel_id,
+			Channel { agent_id, para_id: AssetHubRococo::para_id() },
+		);
+		let versioned_asset_id: VersionedLocation = asset_id.into();
+		snowbridge_pallet_system::Tokens::<Runtime>::insert(token_id, versioned_asset_id);
+	});
+
+	AssetHubRococo::execute_with(|| {
+		type RuntimeOrigin = <AssetHubRococo as Chain>::RuntimeOrigin;
+
+		let assets = vec![Asset { id: AssetId(Location::parent()), fun: Fungible(ROC_AMOUNT) }];
+		let multi_assets = VersionedAssets::V4(Assets::from(assets));
+
+		let destination = VersionedLocation::V4(Location::new(
+			2,
+			[GlobalConsensus(Ethereum { chain_id: CHAIN_ID })],
+		));
+
+		let beneficiary = VersionedLocation::V4(Location::new(
+			0,
+			[AccountKey20 { network: None, key: ETHEREUM_DESTINATION_ADDRESS.into() }],
+		));
+
+		// Send the ROC to Ethereum
+		<AssetHubRococo as AssetHubRococoPallet>::PolkadotXcm::reserve_transfer_assets(
+			RuntimeOrigin::signed(AssetHubRococoReceiver::get()),
+			Box::new(destination),
+			Box::new(beneficiary),
+			Box::new(multi_assets),
+			0,
+		)
+		.unwrap();
+	});
+
+	BridgeHubRococo::execute_with(|| {
+		type RuntimeEvent = <BridgeHubRococo as Chain>::RuntimeEvent;
+
+		type Converter = <bridge_hub_rococo_runtime::Runtime as snowbridge_pallet_inbound_queue::Config>::MessageConverter;
+
+		assert_expected_events!(
+			BridgeHubRococo,
+			vec![
+				RuntimeEvent::EthereumOutboundQueue(snowbridge_pallet_outbound_queue::Event::MessageQueued {..}) => {},
+			]
+		);
+
+		let message_id: H256 = [0; 32].into();
+		let message = VersionedMessage::V1(MessageV1 {
+			chain_id: CHAIN_ID,
+			command: Command::SendNativeToken {
+				token_id,
+				destination: Destination::ForeignAccountId32 {
+					para_id: AssetHubRococo::para_id().into(),
+					id: AssetHubRococoReceiver::get().into(),
+					fee: XCM_FEE,
+				},
+				amount: ROC_AMOUNT,
+			},
+		});
+		let (xcm, _) = Converter::convert(message_id, message).unwrap();
+		let _ = EthereumInboundQueue::send_xcm(xcm, AssetHubRococo::para_id().into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubRococo,
+			vec![
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},
+			]
+		);
+	});
+
+	AssetHubRococo::execute_with(|| {
+		type RuntimeEvent = <AssetHubRococo as Chain>::RuntimeEvent;
+
+		// Check that the token was received and issued as a foreign asset on AssetHub
+		assert_expected_events!(
+			AssetHubRococo,
+			vec![
+				RuntimeEvent::Balances(pallet_balances::Event::Minted { .. }) => {},
```

Comment:
The mint is weird for me here, would have thought the DOT would be locked on AH or something, and then unlocked when it is sent back? Can you perhaps explain what happens on AH when the native token is sent to Ethereum?

Reply:
Yean, that's true. Native assets will be locked on the source chain and minted on the destination chain.
For the reverse direction as a foreign asset, it will be burnt on the source chain and then unlocked on the destination chain.

Reply:
Nice, thanks! That clears it up for me.